### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend function Relu-6, 13, 14 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20564]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of Relu for next list of opsets: [opset 6](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-6),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-13),  [opset 14](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-14)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Relu)
More details can be found in ONNX Changelog: [opset 6](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-6),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-13),  [opset 14](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Relu-14)
1. Function already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by p-wysocki** [comment]
.take

**Comment by Aryan8912** [comment]
please assign to me so I can working on this issue

**Comment by Aryan8912** [comment]
please give me file location to me

**Comment by p-wysocki** [comment]
Hello @Aryan8912, please start with the "What needs to be done" section, together with Contribution Guide from "Resources" section. :)

**Comment by Aryan8912** [comment]
Can't find it in discussions brother

**Comment by p-wysocki** [comment]
Hello @Aryan8912, I'm afraid I don't quite understand your question. The task regards finding out what the differences between different Relu implementations are and fixing them. The 1st point of TODO list contains: "Function already has a common implementation in OpenVINO."

Are you looking for the file where Relu is implemented in OpenVINO?

**Comment by Aryan8912** [comment]
Yes brother

**Comment by p-wysocki** [comment]
I believe @gkrivor meant OpenVINO implementation (as opposed to ONNX Frontend implementation). Relu can be found under https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/relu.cpp.

**Comment by p-wysocki** [comment]
Hi @Aryan8912, are you still working on the task?

**Comment by Aryan8912** [comment]
Yes please give me some time bro

**Comment by Aryan8912** [comment]
hey Sir, I have made a pull request , I am not able to understand what is the problem with this PR of mine. please help me out

**Comment by RitikaxShakya** [comment]
Hello, is this issue open? if so can i take up this issue?

**Comment by mlukasze** [comment]
let's give a chance to current assignee, but I will remember about you

**Comment by AJThePro99** [comment]
.take

**Comment by janviisonii23** [comment]
Is someone working on this issue ? If not can I work on this issue? @mlukasze

**Comment by mlukasze** [comment]
hey @AJThePro99 are you making any progress? do you need support maybe? or you don't have time to continue?

**Comment by AJThePro99** [comment]
> hey @AJThePro99 are you making any progress? do you need support maybe? or you don't have time to continue?

Hi, I was intending to work on this issue. But the current issue #20547 that I'm working on, my progress was delayed due to dependency issues.
It's back on track, and I expect it to be merged after I open a new PR.

You can unassign me from this issue 👍

**Comment by mlukasze** [comment]
thanks for quick update!
you are right, let's focus on one ticket at the time, there will be planty more later :)

@janviisonii23 assigning this one to you

**Comment by thesakshidiggikar** [comment]
Hey I Would like to solve this issue , as i went through repo. could you assign me this task?

**Comment by mlukasze** [comment]
hey @thesakshidiggikar 
it depends on @janviisonii23 who is current assignee and working on it.
Maybe this one would fit you as well? => https://github.com/openvinotoolkit/openvino/issues/20553

**Comment by thesakshidiggikar** [comment]
@mlukasze superb ! thanks for suggesting I would like to work on it can you assign me that?

**Comment by p-wysocki** [comment]
Hello @janviisonii23, let's settle the task status - are you still on it? Do you require any assistance?

**Comment by Phantom65536** [comment]
.take

**Comment by Phantom65536** [comment]
I've created test cases and I tried running them with `/bin/intel64/Debug/ov_onnx_frontend_tests --gtest_filter=*onnx_model_relu_*`. @gkrivor said "OpenVINO test infrastructure then converts prototxt files to ONNX models" but I find no onnx models under "/scratch/chanelsi/openvino/bin/intel64/Debug/./test_model_zoo/onnx/<name>.onnx", which is the path convert_model() in each test in onnx_import.in.cpp is looking for. I understand the . in the path may cause the issue, but there's indeed no converted onnx files in the entire workspace.

Here's my build option (run on build directory):
```
cmake -DCMAKE_BUILD_TYPE=Debug -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_FUNCTIONAL_TESTS=ON -DENABLE_DEBUG_CAPS=ON -DENABLE_CPU_DEBUG_CAPS=ON -DENABLE_NCC_STYLE=ON ..
```

There're no onnx models converted after running `cmake --build . --parallel`

**Comment by AJThePro99** [comment]
> I've created test cases and I tried running them with `/bin/intel64/Debug/ov_onnx_frontend_tests --gtest_filter=*onnx_model_relu_*`. [@gkrivor](https://github.com/gkrivor) said "OpenVINO test infrastructure then converts prototxt files to ONNX models" but I find no onnx models under "/scratch/chanelsi/openvino/bin/intel64/Debug/./test_model_zoo/onnx/.onnx", which is the path convert_model() in each test in onnx_import.in.cpp is looking for. I understand the . in the path may cause the issue, but there's indeed no converted onnx files in the entire workspace.
> 
> Here's my build option (run on build directory):
> 
> ```
> cmake -DCMAKE_BUILD_TYPE=Debug -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_FUNCTIONAL_TESTS=ON -DENABLE_DEBUG_CAPS=ON -DENABLE_CPU_DEBUG_CAPS=ON -DENABLE_NCC_STYLE=ON ..
> ```
> 
> There're no onnx models converted after running `cmake --build . --parallel`

Hello, I too faced the same issue one point in time, but I found the solution. It's quite a simple fix. 
It's simply because you haven't installed the python packages needed for generating ONNX models.
___
There is a [requirements.txt](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/requirements.txt) in `<project-dir>/src/frontends/onnx/tests`. 
Use pip to install the requirements `pip install -r requirements.txt` in your preferred manner.
And next time you run the build, `.onnx` files will be generated.

Hope this solves your issue

**Comment by Phantom65536** [comment]
I have made the requested changes. Thanks for reviewing.

**Comment by goyaladitya05** [comment]
.take

**Comment by goyaladitya05** [comment]
Hey @mlukasze, 
I’m working on this issue but wasn't able to tag @gkrivor on the PR. Since the issue has been open for a while, could you please let me know who is currently overseeing the ONNX Frontend or if I should just proceed with the PR for your review? Thanks.
